### DISCUSSION TITLE
assertion error while generating preview on NextCloud

### DISCUSSION THREAD

**Discussion by jhanyung** [ImageMagick/ImageMagick#3353] [Help]
Hi, I'm using ImageMagick to generate preview for NextCloud gallery.

I'm just starting out, so I might not understand all your answers, sorry!

Here's the command I ran
`su -m www -c 'php occ preview:generate-all`

I got very very far, around 2, 3 hours into the command when in one of the directories i get:

`Assertion failed: (string_info->signature == MagickCoreSignature), function GetStringInfoDatum, file MagickCore/string.c, line 1208.
Abort`

I ran again and thankfully it skips the ones already been generated.
However it does stop again at the same spot.

What can i do at this point?

Thanks for the help!!

**Comment by snibgo** [comment]
First question: what version of ImageMagick is being used? If it is old, I suggest you upgrade it and try again.

If that doesn't help: what ImageMagick command (or function) is being executed, and what is the input image? If we know that, we can try to track down an IM bug, or declare that your input file is corrupt, or whatever.
